File: Testing_Catalog/CAlgorithms/Search/IntTarget/linearSearch.c
//================================================
//================================================
// Linear Search
// linearSearch.c
//
// Isaac A.B. Yep
//
// Time Complexity:
//
// Space Complexity:
//
//================================================
//================================================
#include "main.h"

// algorithm
int linear_search(int* array, int size, int target)
{
    for (int i = 0; i < size; ++i)
        if (array[i] == target)
            return i;
    return -1;
}

File: Testing_Catalog/PythonAlgorithms/Search/IntTarget/linearSearch.py
'''
Linear Search

Runtime Complexity within O(n)

takes list of integers and integer to find, returns index of target or -1 if not found
'''

def linear_search(array: list, target: int) -> int:

    for i in range(len(array)):
        if array[i] == target:
            return i;
    return -1

